src/tests/decider-distro-tests-rs/tests/test.rs

```diff
@@ -48,15 +75,402 @@ pub fn make_distro() -> PackageDistro {
+        worker_period_sec: 0,
+        worker_ipfs_multiaddr: "/ip4/127.0.0.1/tcp/5001".to_string(),
+        chain_api_endpoint: "http://127.0.0.1:12009".to_string(),
+        chain_network_id: 11,
+        chain_contract_block_hex: "0x0".to_string(),
+        chain_matcher_addr: "0x0".to_string(),
+        chain_workers_gas: 210_00,
+        chain_wallet_key: "0x0".to_string(),
+    };
+    // let's try to run a decider cycle on demand by updating the config
+    let mut trigger_config = TriggerConfig::default();
+    trigger_config.clock.start_sec = 1;
+    make_distro(trigger_config, decider_settings)
+}
+
+pub fn make_distro_with_api(api: String) -> PackageDistro {
+    let decider_settings = DeciderConfig {
+        // worker will run once
+        worker_period_sec: 0,
+        worker_ipfs_multiaddr: "/ip4/127.0.0.1/tcp/5001".to_string(),
+        chain_api_endpoint: api,
+        chain_network_id: 11,
+        chain_contract_block_hex: "0x0".to_string(),
+        chain_matcher_addr: "0x0".to_string(),
+        chain_workers_gas: 210_00,
+        chain_wallet_key: "0xfdc4ba94809c7930fe4676b7d845cbf8fa5c1beae8744d959530e5073004cf3f"
+            .to_string(),
+    };
+    // decider will run once
+    let mut trigger_config = TriggerConfig::default();
+    make_distro(trigger_config, decider_settings)
+}
+
+pub fn make_distro_with_api_and_config(api: String, config: TriggerConfig) -> PackageDistro {
+    let decider_settings = DeciderConfig {
+        // worker will run once
+        worker_period_sec: 0,
+        worker_ipfs_multiaddr: "/ip4/127.0.0.1/tcp/5001".to_string(),
+        chain_api_endpoint: api,
+        chain_network_id: 11,
+        chain_contract_block_hex: "0x0".to_string(),
+        chain_matcher_addr: "0x0".to_string(),
+        chain_workers_gas: 210_00,
+        chain_wallet_key: "0xfdc4ba94809c7930fe4676b7d845cbf8fa5c1beae8744d959530e5073004cf3f"
+            .to_string(),
+    };
+    // decider will run once
+    make_distro(config, decider_settings)
+}
+
+async fn execute(
+    client: &mut ConnectedClient,
+    correct_air: &str,
+    return_values: &str,
+    mut data: HashMap<&str, Value>,
+) -> eyre::Result<Vec<Value>> {
+    data.insert("relay", json!(client.node.to_string()));
+    data.insert("client", json!(client.peer_id.to_string()));
+
+    client
+        .execute_particle(
+            format!("(seq {correct_air} (call client (\"return\" \"\") [{return_values}]) )"),
+            data,
+        )
+        .await
+}
+
+async fn update_config(
+    client: &mut ConnectedClient,
+    trigger_config: &TriggerConfig,
+) -> eyre::Result<Vec<Value>> {
+    execute(
+        client,
+        r#"(call relay ("spell" "update_trigger_config") ["decider" config])"#,
+        "\"done\"",
+        hashmap! {
+            "config" => json!(trigger_config),
+        },
+    )
+    .await
+}
+
+fn enable_decider_logs() {
+    let namespaces = vec![
+        "run-console=debug",
+        // "spell_event_bus=trace",
```

Comment:
let's clean up that a little bit. I think it's OK to have commented lines, but I'd wish them more consistently formatted